pkg/decoders/base64.go

```diff
@@ -26,51 +26,88 @@ func (d *Base64) FromChunk(chunk *sources.Chunk) *sources.Chunk {
 	}
 
 	if len(decodedSubstrings) > 0 {
-		for substring, dec := range decodedSubstrings {
-			chunk.Data = bytes.ReplaceAll(chunk.Data, []byte(substring), dec)
+		var result bytes.Buffer
+		result.Grow(len(chunk.Data))
+
+		start := 0
+		for _, encoded := range encodedSubstrings {
+			if decoded, ok := decodedSubstrings[encoded]; ok {
+				end := bytes.Index(chunk.Data[start:], []byte(encoded))
+				if end != -1 {
+					result.Write(chunk.Data[start : start+end])
+					result.Write(decoded)
+					start += end + len(encoded)
+				}
+			}
 		}
+		result.Write(chunk.Data[start:])
+		chunk.Data = result.Bytes()
 		return chunk
 	}
 
 	return nil
 }
 
-func getSubstringsOfCharacterSet(data []byte, charset []byte, threshold int) []string {
-	count := 0
-	var substrings []string
-	letters := bytes.Buffer{}
-
+func getSubstringsOfCharacterSet(data []byte, threshold int) []string {
 	if len(data) == 0 {
 		return nil
 	}
 
+	// Given characters are mostly ASCII, we can use a simple array to map.
+	var b64CharsetMapping [128]bool
+	// Build an array of all the characters in the base64 charset.
+	for _, char := range b64Charset {
+		b64CharsetMapping[char] = true
+	}
```

Comment:
I'm not sure if it will make much difference, but this could be done once with a package var populated with an init() function, which automatically gets called once when the package is imported.

Reply:
I think that makes sense. Updated.